python/hsfs/core/opensearch.py

```diff
@@ -16,15 +16,75 @@
 
 import logging
 import re
+from functools import wraps
 
+import urllib3
 from hsfs import client
 from hsfs.client.exceptions import FeatureStoreException, VectorDatabaseException
 from hsfs.core.opensearch_api import OpenSearchApi
+from opensearchpy import OpenSearch
+from opensearchpy.exceptions import (
+    AuthenticationException,
+    ConnectionError,
+    ConnectionTimeout,
+    RequestError,
+)
+from retrying import retry
+
+
+def _is_timeout(exception):
+    return (isinstance(exception, urllib3.exceptions.ReadTimeoutError)
+            or isinstance(exception, ConnectionTimeout))
+
+def _handle_opensearch_exception(func):
+    @wraps(func)
+    def error_handler_wrapper(*args, **kw):
+        try:
+            return func(*args, **kw)
+        except Exception as e:
+            if _is_timeout(e):
+                raise FeatureStoreException(OpenSearchClientSingleton.TIMEOUT_ERROR_MSG) from e
+            else:
+                raise e
+
+    return error_handler_wrapper
+
+class OpensearchRequestOption:
+    TIMEOUT = "timeout"
+
+    @classmethod
+    def get_options(cls, options: dict):
+        """
+        Construct a map of options for the request to the vector database.
+
+        Args:
+            options (dict): The options used for the request to the vector database.
+                The keys are attribute values of the OpensearchRequestOption class.
+
+        Returns:
+            dict: A dictionary containing the constructed options map, where keys represent
+            attribute values of the OpensearchRequestOption class, and values are obtained
+            either from the provided options or default values if not available.
+        """
+        option_map = {
+            cls.TIMEOUT: 30,
+        }
+        if options:
+            # Iterate over attributes of the OpensearchRequestOption class
+            for _, attr_value in vars(cls).items():
+                # Check if the attribute value exists in the options dictionary
+                if attr_value in options:
+                    option_map[attr_value] = options[attr_value]
+        return option_map
```

Comment:
i dont like that we define variable (in this case timeout) `name` and `value` separately. also this could cause confusion if user provides keys with wrong case.

might be easier to work with if done like this:
```
class OpensearchRequestOption:
    OPTION_MAP = {
        "timeout": 30,
    }

    @classmethod
    def get_options(cls, options):
        # make lower case to avoid issues with cases
        options =  {k.lower(): v for k, v in options.items()}
        
        new_options = {}
        if options:
            for attr_name, attr_value in cls.OPTION_MAP.items():
                if attr_name in options:
                    new_options[attr_name] = options[attr_name]
                else:
                    new_options[attr_name] = attr_value
        return new_options
```